assets/src/edit-story/components/richText/styleManipulation.js

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Modifier, EditorState } from 'draft-js';
+
+/**
+ * Internal dependencies
+ */
+import {
+  weightToStyle,
+  styleToWeight,
+  NONE,
+  WEIGHT,
+  ITALIC,
+  UNDERLINE,
+  SMALLEST_BOLD,
+  DEFAULT_BOLD,
+} from './customConstants';
+
+function getPrefixStyleForCharacter(styles, prefix) {
+  const list = styles.toArray().map((style) => style.style ?? style);
+  if (!list.some((style) => style && style.startsWith(prefix))) {
+    return NONE;
+  }
+  return list.find((style) => style.startsWith(prefix));
+}
+
+function getPrefixStylesInSelection(editorState, prefix) {
+  const selection = editorState.getSelection();
+  const styles = new Set();
+  if (selection.isCollapsed()) {
+    styles.add(
+      getPrefixStyleForCharacter(editorState.getCurrentInlineStyle(), prefix)
+    );
+    return [...styles];
+  }
+
+  const contentState = editorState.getCurrentContent();
+  let key = selection.getStartKey();
+  let startOffset = selection.getStartOffset();
+  const endKey = selection.getEndKey();
+  const endOffset = selection.getEndOffset();
+  let hasMoreRounds = true;
+  while (hasMoreRounds) {
+    hasMoreRounds = key !== endKey;
+    const block = contentState.getBlockForKey(key);
+    const offsetEnd = hasMoreRounds ? block.getLength() : endOffset;
+    const characterList = block.getCharacterList();
+    for (
+      let offsetIndex = startOffset;
+      offsetIndex < offsetEnd;
+      offsetIndex++
+    ) {
+      styles.add(
+        getPrefixStyleForCharacter(
+          characterList.get(offsetIndex).getStyle(),
+          prefix
+        )
+      );
+    }
+    if (!hasMoreRounds) {
+      break;
+    }
+    key = contentState.getKeyAfter(key);
+    startOffset = 0;
+  }
+
+  return [...styles];
+}
```

Comment:
This one is a mix of other functions doing the same thing - iterating over all selected characters across whichever blocks and styles they're in. See [this](https://github.com/webdeveloperpr/draft-js-custom-styles/blob/f3e6b533905de8eee6da54f9727b5e5803d53fc4/src/index.js#L8-L52) and [this](https://github.com/facebook/draft-js/issues/602#issuecomment-584676405) for reference.